src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -1,23 +1,39 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
 {
     [System.ComponentModel.DesignerCategory("code")] // this must be fully qualified
     internal abstract class AbstractOptionPage : UIElementDialogPage
     {
-        protected abstract AbstractOptionPageControl CreateOptionPage(IServiceProvider serviceProvider);
+        private static IOptionService s_optionService;
+        private static OptionStore s_optionStore;
+        private static bool s_needsLoadOnNextActivate = true;
+
+        protected abstract AbstractOptionPageControl CreateOptionPage(IServiceProvider serviceProvider, OptionStore optionStore);
 
         protected AbstractOptionPageControl pageControl;
-        private bool _needsLoadOnNextActivate = true;
 
         private void EnsureOptionPageCreated()
         {
+            if (s_optionStore == null)
+            {
+                var componentModel = (IComponentModel)this.Site.GetService(typeof(SComponentModel));
+                var workspace = componentModel.GetService<VisualStudioWorkspace>();
+                s_optionService = workspace.Services.GetService<IOptionService>();
```

Comment:
I'm guessing this should be directly importing `IGlobalOptionService`? The comment does say:

https://github.com/dotnet/roslyn/blob/1e3585793758cafd42ca9ba7bda1eab88da681d6/src/Workspaces/Core/Portable/Options/IGlobalOptionService.cs#L13-L14

But I think you are the exception in this place. Usually the assumption is people shouldn't be reading global options _directly_ because a workspace can provide customized options. But in this case this is _configuring_ the global options themselves.